src/modinfodialogconflicts.cpp

```diff
@@ -886,82 +886,113 @@ std::optional<ConflictItem> AdvancedConflictsTab::createItem(
 
   std::wstring before, after;
 
+  auto currOrigin = m_tab->origin();
+  bool isCurrOrigArchive = archive;
+
   if (!alternatives.empty()) {
     const bool showAllAlts = ui->conflictsAdvancedShowAll->isChecked();
 
     int beforePrio = 0;
     int afterPrio = std::numeric_limits<int>::max();
 
-    for (const auto& alt : alternatives)
-    {
-      const auto& altOrigin = ds.getOriginByID(alt.first);
-
+    if (currOrigin->getID() == fileOrigin) {
+      // current origin is the active winner, all alternatives go in 'before'
+      
       if (showAllAlts) {
-        // fills 'before' and 'after' with all the alternatives that come
-        // before and after this mod in terms of priority
-
-        if (altOrigin.getPriority() < m_tab->origin()->getPriority()) {
-          // add all the mods having a lower priority than this one
+        for (const auto& alt : alternatives)
+        {
+          const auto& altOrigin = ds.getOriginByID(alt.first);
           if (!before.empty()) {
             before += L", ";
           }
 
           before += altOrigin.getName();
-        } else if (altOrigin.getPriority() > m_tab->origin()->getPriority()) {
-          // add all the mods having a higher priority than this one
+        }
+      }
+      else {
+        // only add nearest, which is the last element of alternatives
+        const auto& altOrigin = ds.getOriginByID((alternatives.end()-1)->first);
+
+        before += altOrigin.getName();
+      }
+
+    }
+    else {
+      // current mod is one of the alternatives, find it's position
+      
+      auto currOrgId = currOrigin->getID();
+
+      auto currModIter = std::find_if(alternatives.begin(), alternatives.end(),
+        [&currOrgId](auto const& alt) {
+          return currOrgId == alt.first;
+      });
+
+      if (currModIter != alternatives.end()) {
+        isCurrOrigArchive = currModIter->second.first.size() > 0;
```

Comment:
We should really use a proper struct for this. In the meantime, this looks like it's checking for the size of some container, but it's just a string for the archive name. I'd suggest `currModIter->second.first != ""` for now.

Reply:
Yeah I strongly agree on changing it to a proper struct. It would really help for readability. Right now it's a mess. I'd rather leave it with size() for now since that is how the rest of the code currently does it anywhere else so if I have to go and fix it up later it will be easier: https://github.com/ModOrganizer2/modorganizer/blob/2833f8d61b56a3f4ed6b2a8fafeb679033e6989e/src/shared/fileentry.cpp#L97